skbio/metadata/_interval.py

```diff
@@ -0,0 +1,274 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
```

Comment:
there is no name clashing. I am thinking of `Mapping` just because of the API conformity. It is not a question of must-have or must-not-have. either way is fine.